src/Okta.Sdk.IntegrationTest/OAuthScenarios.cs

```diff
@@ -953,7 +953,99 @@ public async Task ListUsersWithPaginationAndRetry()
+                                ],
+                                ""token_endpoint_auth_method"": ""private_key_jwt"",
+                                ""application_type"": ""service"",
+                                ""jwks"": {{
+                                    ""keys"": [
+                                        {{
+                                            ""kty"":""RSA"",
+                                            ""e"":""AQAB"",
+                                            ""n"":""mTjMc8AxU102LT1Jf-1qkGmaSiK4L7DDlC1SMvtyCRbDaiJDIagedfp1w8Pgud8YWOaS5FFx0S6JqGGP2U8OtpowzBcv5sYa-e5LHfnoueTJPj_jnI3fj5omZM1w-ofhFLPZoYEQ7DFYw0yLrzf8zaKB5-9BZ8yyOLhSKqxaOl2s7lw2TrwBRuQpPXmEir70oDPvazd8-An5ow6F5q7mzMtHAt61DJqrosRHiRwh4N37zIX_RNu-Tn1aMktCBl01rdoDyVq7Y4iwNH8ZAtT5thKK2eo8d-jb9TF9PH6LGffYCth157w-K4AZwXw74Ybo5NOux3XpIpKRbFTwvBLp1Q""
+                                        }}   
+                                   ]
+                                }}
+                             }}";
+            
+            var oktaDomain = Configuration.GetConfigurationOrDefault().OktaDomain;
+
+            // Remove '/' at the end since endpoint fails otherwise
+            if (oktaDomain.EndsWith($"/"))
+            {
+                oktaDomain = oktaDomain.Remove(oktaDomain.Length - 1);   
+            }
+            
+            var apiClient = new ApiClient(oktaDomain);
+            var usersReadGrantPayload = $@"{{
+                                        ""scopeId"" : ""okta.users.read"",
+                                        ""issuer"" : ""{oktaDomain}""
+                                    }}";
+            var appsReadGrantPayload = $@"{{
+                                        ""scopeId"" : ""okta.apps.read"",
+                                        ""issuer"" : ""{oktaDomain}""
+                                    }}";
+            var requestOptions = GetBasicRequestOptions();
+            requestOptions.Data = JObject.Parse(payload);
+
+            var serviceResponse = await apiClient.PostAsync<JObject>("/oauth2/v1/clients", requestOptions);
+            output.WriteLine("Create client response {0}", serviceResponse.Data.ToString());
+            var clientId = serviceResponse.Data["client_id"].ToString();
+
+            try
+            {
+
+                requestOptions = GetBasicRequestOptions();
+                requestOptions.Data = JObject.Parse(usersReadGrantPayload);
+
+                // Add users.read scope grant to the service
+                var usersReadGrantResponse = await apiClient.PostAsync<JObject>($"/api/v1/apps/{clientId}/grants", requestOptions);
+
+                requestOptions = GetBasicRequestOptions();
+                requestOptions.Data = JObject.Parse(appsReadGrantPayload);
 
+                // Add apps.read scope grant to the service
+                var appsReadGrantResponse = await apiClient.PostAsync<JObject>($"/api/v1/apps/{clientId}/grants", requestOptions);
```

Comment:
We should ensure that the grant responses succeed before continuing.

Reply:
Can we also add a comment on how we're structuring this test? It seems we first test individual grants separately before testing multiple grants. It will be helpful to have some comments if we have to revisit this in the future.